src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,695 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Represents a data structure in which each element additionally has a priority
+    /// associated with it.
+    /// </summary>
+    /// <typeparam name="TElement">The type of the element.</typeparam>
+    /// <typeparam name="TPriority">The type of the priority.</typeparam>
+    public class PriorityQueue<TElement, TPriority>
+    {
+        /// <summary>
+        /// Represents an implicit heap-ordered complete d-ary tree, stored as an array.
+        /// </summary>
+        private (TElement element, TPriority priority)[] _nodes;
+
+        private Lazy<UnorderedItemsCollection> _unorderedItems;
+
+        /// <summary>
+        /// The number of nodes in the heap.
+        /// </summary>
+        private int _size;
+
+        /// <summary>
+        /// Used to keep enumerator in sync with the collection.
+        /// </summary>
+        private int _version;
+
+        private const int MinimumGrow = 4;
+        private const int GrowFactor = 200; // double each time
```

Comment:
I had to check the relevant method to understand that `200` means 200%. Why not just `2`?

Reply:
I agree this is not nice, I chose this way to be consistent with `Queue<T>`, which has this as well:

```
private const int MinimumGrow = 4;
private const int GrowFactor = 200;  // double each time
```

Do we want to inconsistent in this case?

Reply:
> Do we have your blessing to be inconsistent in this case?

Yes, my main question however is why would we need to lift this to a const? If it's only being used in one method, from a code readability perspective it's probably more beneficial to just inline value.

Reply:
Usage of magic numbers in code is an anti-pattern, they should be represented with self-documenting constants.

Reply:
We use magic numbers where their meaning is obvious and they are used in a single place. Eg., for
growing list https://source.dot.net/#System.Private.CoreLib/List.cs,403
growing dictionary https://source.dot.net/#System.Private.CoreLib/HashHelpers.cs,78
growing arraylist https://source.dot.net/#System.Private.CoreLib/ArrayList.cs,314

etc. 

Indeed you self document them with `// double each time`

Reply:
The difference with `Queue<T>` is that the value there is dynamic - provided in a constructor. Not so here.

Reply:
Ah I was looking at Queue, not Queue<T>. I think Queue<T> is an outlier. It should have used 2 inline instead of a const 200 that gets divided.